core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -231,6 +225,16 @@ private Object handleTxCommand(TxInvocationContext ctx, TransactionBoundaryComma
          }
       } else {
          if (currentTopologyId() > command.getTopologyId()) {
+            //in some cases, the node will not retry the transaction in this node and then it leaks.
+            //if it is safe, we rollback the transaction. If the coordinator retries on this node, no side effect is created.
```

Comment:
hmm, IMO, this way is safer and I don't had to worry about losing or partition when the command is retried.